cmake/AwsPrebuildDependency.cmake

```diff
@@ -75,3 +92,52 @@ function(aws_prebuild_dependency)
         DESTINATION ${CMAKE_INSTALL_PREFIX}
     )
 endfunction()
+
+# Get list of optional or platform-specific variables that may affect build process.
+function(aws_get_variables_for_prebuild_dependency AWS_CMAKE_PREBUILD_ARGS)
+    set(variables "")
+    set(variablesToIgnore CMAKE_INSTALL_PREFIX)
+
+    # The CMake variables below were chosen for Linux, BSD, and Android platforms. If you want to use the prebuild logic
+    # on other platforms, the chances are you have to handle additional variables (like CMAKE_OSX_SYSROOT). Refer to
+    # https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html to update the list of handled variables, and
+    # then you can enable a new platform here.
+    if (NOT UNIX OR APPLE)
+        message(FATAL_ERROR "aws_get_variables_for_prebuild_dependency is called for unsupported platform")
+    endif()
+
+    get_cmake_property(vars CACHE_VARIABLES)
+    foreach(var ${vars})
+        # Variables in this block make sense only in cross-compiling mode. The variable list is created from the CMake
+        # documentation on toolchains: https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html
+        # NOTE: Some variables are missed here (e.g. CMAKE_SYSROOT) because they can be set via toolchain file only.
+        if (CMAKE_CROSSCOMPILING AND (
+                var STREQUAL "CMAKE_TOOLCHAIN_FILE"
+                OR var STREQUAL "CMAKE_SYSTEM_NAME"
+                OR var STREQUAL "CMAKE_SYSTEM_VERSION"
+                OR var STREQUAL "CMAKE_SYSTEM_PROCESSOR"
+                # Android-specific variables.
+                OR var MATCHES "^(CMAKE_)?ANDROID_"))
+            set(escaped_var ${${var}})
+            # To store a list within another list, it needs to be escaped first.
+            string(REPLACE ";" "\\\\;" escapedVar "${${var}}")
+            list(APPEND variables "-D${var}=${escapedVar}")
+        endif()
+
+        # Other optional variables applicable both in cross-compiling and non-cross-compiling modes.
+        if (var STREQUAL "CMAKE_C_COMPILER"
+                OR var STREQUAL "CMAKE_CXX_COMPILER"
+                OR var STREQUAL "CMAKE_C_FLAGS"
```

Comment:
Does AWS-LC use C++ anywhere? If so, we also need CMAKE_CXX_FLAGS. If not, maybe remove CMAKE_CXX_COMPILER?

Reply:
Oh man. Should we also worry about linker flags? https://cmake.org/cmake/help/latest/envvar/LDFLAGS.html

Reply:
Lots of food for thought in this doc:
https://cmake.org/cmake/help/latest/manual/cmake-env-variables.7.html

Reply:
Good catch with linker flags, I added them.
As for env vars, a cmake process called via `execute_process` sees them all (at least on Linux machines). afaics the usual logic for env vars in cmake is to initialize a property or another variable from them (e.g. LDFLAGS sets `CMAKE_*_LINKER_FLAGS_INIT`). So, we either pass these variables explicitly or cmake will initialize them from env vars.
I hope, I'm not missing anything.